accounts/tests.py

```diff
@@ -1,47 +1,162 @@
+from django.urls import reverse
 from django.test import TestCase
-
+from .models import User
 
 class TestSignUpView(TestCase):
     def test_success_get(self):
-        pass
+        response = self.client.get(reverse('accounts:signup'))
+        self.assertEqual(response.status_code,200)
+        self.assertTemplateUsed(response, 'accounts/signup.html')
 
     def test_success_post(self):
-        pass
+        post ={
+            'email' : 'test@example.com',
+            'username' : 'test',
+            'password1' : 'goodpass',
+            'password2' : 'goodpass',
+        }
+        response = self.client.post(reverse('accounts:signup'), post)
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(User.objects.filter(username = 'test',email = 'test@example.com').exists())#ユーザーが追加されたかを確認　ついでに正しく登録されているかを確認
 
     def test_failure_post_with_empty_form(self):
-        pass
+        post ={
+            'email' : '',
+            'username' : '',
+            'password1' : '',
+            'password2' : '',
+        }
+        response = self.client.post(reverse('accounts:signup'),post)
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(User.objects.filter(username = '',email = '').exists())#追加されていないことを確認
+        self.assertFormError(response, 'form', 'email', 'このフィールドは必須です。')
+        self.assertFormError(response, 'form', 'username', 'このフィールドは必須です。')
+        self.assertFormError(response, 'form', 'password1', 'このフィールドは必須です。')
+        self.assertFormError(response, 'form', 'password2', 'このフィールドは必須です。')
+
 
     def test_failure_post_with_empty_username(self):
-        pass
+        post ={
+            'email' : 'test@example.com',
+            'username' : '',
+            'password1' : 'goodpass',
+            'password2' : 'goodpass',
+        }
+        response = self.client.post(reverse('accounts:signup'), post)
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(User.objects.filter(username = '',email = 'test@example.com').exists())#追加されていないことを確認
+        self.assertFormError(response, 'form', 'username', 'このフィールドは必須です。')
 
     def test_failure_post_with_empty_email(self):
-        pass
+        post ={
+            'email' : '',

```

Comment:
```suggestion
            'email' : ' ',

```
このようにスペース開けた場合でもきエラー出ますか？

Reply:
password1について空白を入れるとエラーが出なかったのですが、これは問題ないでしょうか？
また、password2に空白を入れると「8文字以上にしてください」という旨のエラーが出たのですが、これも問題ないでしょうか？
この2つに関しては空白を入れていない状態なので、空白を入れる必要があれば再度修正して提出します。